app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def add
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new( name: params[:task][:name], description: params[:task][:description], date: params[:task][:date], completion_date: nil )
+    if @task.save
+      redirect_to root_path
+    else
+      render :add
+    end
+  end
+
+  def show
+    @task = Task.find( params[:id].to_i )
```

Comment:
I would suggest having some sort of if statement here for if the task doesn't exist and redirecting them when that happens.  The same would apply for edit, create, update and delete.